server/channels/store/sqlstore/post_store.go

```diff
@@ -221,7 +221,7 @@ func (s *SqlPostStore) SaveMultiple(rctx request.CTX, posts []*model.Post) ([]*m
 		return nil, -1, errors.Wrap(err, "post_tosql")
 	}
 
-	transaction, err := s.GetMaster().Beginx()
+	transaction, err := s.GetMaster().BeginXWithIsolation(&sql.TxOptions{Isolation: sql.LevelSerializable})
```

Comment:
Could you share more context about this change?

Reply:
Ah yes, I actually found this while inserting the data. In sqlstore.savePostMultiple, there is `updateThreadsFromPosts` which is run as part of the transaction. This is responsible for upserting a thread. However, the logic is not written in a way that if 2 replies concurrent hit the server, both will try to create a threads entry and one will fail.

Making a stricter isolation level was one way to achieve that. Although I agree if we spend more time to make the query properly upsertable, that would be the better solution. Happy to revert this if you concur.

Reply:
I tend to prefer the *make the query properly upsertable* option and rely on the isolation level only as a last resort, but ultimately up to you.